src/lib.rs

```diff
@@ -16,14 +16,11 @@ extern crate nom;
 #[cfg(feature = "serde")]
 #[macro_use]
 extern crate serde;
-extern crate string_cache;
 #[macro_use]
 extern crate thiserror;
 
 mod errors;
 
-include!(concat!(env!("OUT_DIR"), "/atoms.rs")); // for QualifierKey, FeatureKind
-
```

Comment:
Ah for `gb-io.py` crate it was really helpful to have `QualifierKey` and `FeatureKind` traits, because I could implement conversion to and from Python onto these :sweat_smile:  

Do you think it would be possible to keep them as newtypes? E.g:
```
pub struct FeatureKind(pub Cow<'static, str>);
```

If not it's fine I can add a workaround but I think in terms of API compatibility it may be nice to keep them separated (also in case later you want to try using a `Rc` for storing the keys, idk).

Reply:
Hmm I need to think about this a bit more - I'm leaning towards string for simplicity but could be convinced otherwise. Would it be a lot of work to adapt the python conversion?

Reply:
I can adapt it using wrapper types, I just thought that maybe having `FeatureKind` and `QualifierKey` types that are basically `Deref<str> + From<str>` so that you can abstract the implementation.

Reply:
I had a go at this, the ability to abstract the implementation is appealing, but in the end I think the extra complexity might not be worth it, I'm gonna merge this as is but would still be open to changing it in the future.